src/lava/magma/compiler/compiler.py

```diff
@@ -299,13 +299,42 @@ def _create_subcompilers(
             initialized with the Processes they will compile.
         """
         subcompilers = []
+        c_idx = []
+        nc_idx = []
         # Go through all required subcompiler classes...
-        for subcompiler_class, procs in compiler_type_to_procs.items():
+        for idx, (subcompiler_class, procs) in \
+                enumerate(compiler_type_to_procs.items()):
             # ...create the subcompiler instance...
             compiler = subcompiler_class(procs, self._compile_config)
             # ...and add it to the list.
             subcompilers.append(compiler)
-
+            # Remember the index for C and Nc subcompilers:
+            if isinstance(compiler, CProcCompiler):
+                c_idx.append(idx)
+            if isinstance(compiler, NcProcCompiler):
+                nc_idx.append(idx)
+
+        # Implement the heuristic "C-first Nc-second" whenever C and Nc
+        # compilers are in the list. The subcompiler.compile is called in
+        # their order of appearance in this list.
+        # ToDo: In the implementation below, we assume that there is only one
```

Comment:
That assumption should not be made. The compiler was designed to create multiple instances of the same type of subcompiler, one instance for every group of Processes.

Reply:
Sorry, this is not correct, I didn't see that this method was working on individual ProcGroups already.